bundles/framework/mydata/instance.js

```diff
@@ -0,0 +1,312 @@
+ *
+ * See Oskari.mapframework.bundle.mydata.MyDataBundle for bundle definition.
+ */
+Oskari.clazz.define('Oskari.mapframework.bundle.mydata.MyDataBundleInstance',
+
+    /**
+     * @method create called automatically on construction
+     * @static
+     */
+
+    function () {
+        this.core = null;
+        this.sandbox = null;
+        this.started = false;
+        this.template = null;
+        this.plugins = {};
+        this.viewService = null;
+    }, {
+        /**
+         * @static
+         * @property __name
+         */
+        __name: 'MyData',
+        /**
+         * @method getName
+         * @return {String} the name for the component
+         */
+        'getName': function () {
+            return this.__name;
+        },
+        /**
+         * @method setSandbox
+         * @param {Oskari.Sandbox} sandbox
+         * Sets the sandbox reference to this component
+         */
+        setSandbox: function (sandbox) {
+            this.sandbox = sandbox;
+        },
+        /**
+         * @method getSandbox
+         * @return {Oskari.Sandbox}
+         */
+        getSandbox: function () {
+            return this.sandbox;
+        },
+        /**
+         * @method getViewService
+         * @return {Oskari.mapframework.bundle.mydata.service.ViewService}
+         */
+        getViewService: function () {
+            return this.viewService;
+        },
+        /**
+         * @method getLocalization
+         * Returns JSON presentation of bundles localization data for current language.
+         * If key-parameter is not given, returns the whole localization data.
+         *
+         * @param {String} key (optional) if given, returns the value for key
+         * @return {String/Object} returns single localization string or
+         *      JSON object for complete data depending on localization
+         *      structure and if parameter key is given
+         */
+        getLocalization: function (key) {
+            if (!this._localization) {
+                this._localization = Oskari.getLocalization(this.getName());
+            }
+            if (key) {
+                return this._localization[key];
+            }
+            return this._localization;
+        },
+        /**
+         * @method start
+         * implements BundleInstance protocol start methdod
+         */
+        'start': function () {
+            var me = this;
+
+            if (me.started) {
+                return;
+            }
+
+            me.started = true;
+
+            var conf = this.conf || {};
+            this.conf = conf;
+            var sandboxName = (conf ? conf.sandbox : null) || 'sandbox';
+            var sandbox = Oskari.getSandbox(sandboxName),
+                p;
+
+            me.sandbox = sandbox;
+            this.viewService = Oskari.clazz.create('Oskari.mapframework.bundle.mydata.service.ViewService', Oskari.urls.getRoute());
```

Comment:
ViewService isn't using the constructor param:
```suggestion
            this.viewService = Oskari.clazz.create('Oskari.mapframework.bundle.mydata.service.ViewService');
```